Userland/Libraries/LibDSP/Track.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2021, kleines Filmröllchen <malu.bertsch@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "Clip.h"
+#include "Music.h"
+#include "Processor.h"
+#include <LibCore/Object.h>
+
+namespace LibDSP {
+
+// A track is also known as a channel and serves as a container for the audio pipeline: clips -> processors -> mixing & output
+class Track : public Core::Object {
+    C_OBJECT_ABSTRACT(Track)
+public:
+    Track(NonnullRefPtr<Transport> transport)
+        : m_transport(transport)
+    {
+    }
+    virtual ~Track()
+    {
+    }
+
+    virtual bool check_processor_chain_valid() const = 0;
+    bool add_processor(NonnullRefPtr<Processor> new_processor);
+
+    // Creates the current signal of the track by processing current note or audio data through the processing chain
+    Sample current_signal();
+
+    NonnullRefPtrVector<Processor> processor_chain() const { return m_processor_chain; }
+    const NonnullRefPtr<Transport> transport() const { return m_transport; }
```

Comment:
`NonnullRefPtr<Transport> const`.

Also, for these getter, you want to return raw pointers quite a lot of the time, since it does maybe not make sense if the `this` object is destroyed but the `Transport` is still there. Though I am not sure if this applies here.

Reply:
For Processor sure, because Track "owns" them. For transport, that's a pseudo-global state object of the entire DSP system, so rather not.